### DISCUSSION TITLE
Problem requesting a certificate from proxmox (ACME)

### DISCUSSION THREAD

**Discussion by dishorned** [smallstep/certificates#896] [Q&A] [answered]
Hey Guys, hope you are doing well!

I set up a smallstep-ca to use it as ACME server. Root and Intermediate certificate were generated without a problem. My frist service to try the ACME server was pihole. With certbot, it was no problem to get a certificate for pihole.domain.local. i used the following command:

`_certbot certonly --standalone -d pihole.domain.local --server https://xxx.xxx.xxx.xxx:8443/acme/acme/directory_`

Next thing I tried was a proxmox virtual environement (pve) host. I set up the ACME account on the pve with the following command:

`_pvenode acme account register account-name email-address@domain.local --directory https://xxx.xxx.xxx.xxx:8443/acme/acme/directory_`

It shows up as valid in the pve ACME configuration. Final step should be to order a certificate. I tried it via GUI and on the shell, but both failed showing up the following message on pve-GUI:

`_TASK ERROR: Error: POST to https://xxx.xxx.xxx.xxx:8443/acme/acme/challenge/5n0UCBRcdFIw3esPMxKsqIJjs0ZWH0r7/QS9CHEzLHKYsEilt27pjkx9fpUU2iYdY Status read failed:  at /usr/share/perl5/Net/HTTP/Methods.pm line 282_`

I double checked the /var/log/step-ca/output.log for any information and the only hint I found was the following:

`_urn:ietf:parmas:Acme:error:connection (...) The server could not connect to the validation target_ `

The smallstep and pihole VM are in the same vlan, but the pve-host is in a different one. I allowed the ports 443, 8443 and 80 from the pve-host to the smallstep-server, but it seems, that only 8443 is necessary, because I set it in the smallsetp configuration. Seems that no port to open is necessary from the smallstep server to the pve-host.

Does anyone know, what the problem could be or do you need a specific logfile?

Thanks a lot an best regards

**Comment by maraino** [comment, marked as answer]
Hi @dishorned, I haven't implemented the ACME integration, so I might not have all the details. But I know that with the most typical challenge, HTTP-01, `step-ca` requires access to the port `80` of the client, in this case, your proxmoc. If the `TLS-ALPN-01` challenge is used, then the port `443` is required. The last option is to you your DNS with the DNS-01.

**Reply by maraino** [reply]
From your pve to `step-ca` you only need the port where `step-ca` is listening, but the ACME challenges define the ports the `step-ca` needs to access.

**Reply by dishorned** [reply]
Hi maraino,

first: Thanks a lot for your help and answer.

I checked it three times yesterday, but found no blocked connection in the firewall-log. So I really thought, no open port from step-ca to proxmox is needed. After I read your answer, I checked it again and saw a blocked connection. I wasted several hours yesterday to find out what the problem is, and I think, I only chose the false filter in the firewall log :Z 

From pve-host to step-ca, the configured port should be open.
From step-ca to pve-host, port 80 has to been open.

I think every IT guy happens that at least once in a lifetime. Now it works and I'm happy.

Thanks a lot and have a nice day
